### PR TITLE
Update changelog to 4.0.0

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

**Comment by ndellingwood** [comment]
@lucbv I think some of the bug fix entries can be dropped that were part of new features added this release cycle (that is, fixes for bugs added during this release cycle, not a preexsiting bug). Aside from dropping those, let me know of any changes. I pushed the branch to the kokkos/kokkos-kernels repo so that it can be easily updated by team members

**Comment by lucbv** [comment]
Okay, I will have a look and if I spot something I'll let you know. I can also go through the change log and do some editing as well. Thanks Nathan!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,103 @@
...
+- ROTMG: rocblas TPL turned off [\#1603](https://github.com/kokkos/kokkos-kernels/pull/1603)
+- Fix HIP nightly build on ORNL Jenkins CI server [\#1544](https://github.com/kokkos/kokkos-kernels/pull/1544)
+- Turn on cublas and cusparse in CLANG13CUDA10 CI check [\#1584](https://github.com/kokkos/kokkos-kernels/pull/1584)
+- Add clang13+cuda10 PR build [\#1524](https://github.com/kokkos/kokkos-kernels/pull/1524)
+- .githob/workflows: Fix redundant workflow triggers [\#1527](https://github.com/kokkos/kokkos-kernels/pull/1527)
+- Add GCC test options for C++17 and disable perftests for INTEL19 [\#1511](https://github.com/kokkos/kokkos-kernels/pull/1511)
+- Add INTEL19 and CUDA11 CI settings [\#1505](https://github.com/kokkos/kokkos-kernels/pull/1505)
+- .github/workflows: use c++17 [\#1484](https://github.com/kokkos/kokkos-kernels/pull/1484)
+
+### Bug Fixes:
```
Comment (vqd8a) [inline CHANGELOG.md:68]: @ndellingwood Can you add #1683 here?

Comment (ndellingwood) [reply CHANGELOG.md:68]: Good catch, thanks @vqd8a ! I may have missed a couple of the cherry-picks, will look back over those as well

Comment (ndellingwood) [reply CHANGELOG.md:72]: Added the entry along with other cherry-picks I missed, ed5ab13316bf4a9197264beb6878a2e282e490dc

**Pull request merged**